_tuning-your-cluster/availability-and-recovery/rule-based-autotagging/autotagging.md

```diff
@@ -0,0 +1,197 @@
+2. **Pattern matching**: Supports wildcards (such as index patterns)
+3. **List matching**: Matches any item in a list
+4. **Range matching**: Matches values within a defined range
+
+For example, in workload management, index patterns support:
+
+* Exact match: `logs-2025-04`
+* Prefix pattern: `logs-2025-*`
+
+Note: Matching behavior is determined by the feature and attribute type.
+
+### Rule precedence
+
+When multiple rules match a request, OpenSearch uses the following precedence rules:
+
+1. Rules with more specific attribute matches take priority
+2. Feature-specific tie-breaking logic is applied
+
+For example, with index patterns:
+
+* `logs-prod-2025-*` takes precedence over `logs-prod-*`
+* `logs-prod-*` takes precedence over `logs-*`
+
+### Evaluation process
+
+OpenSearch evaluates incoming requests using the following process:
+
+1. OpenSearch receives a request
+2. The system evaluates request attributes against defined rules
+3. The most specific matching rule’s value is assigned
+4. If no rules match, no value is assigned
+
+## Examples
+
+These examples demonstrate how rule-based auto-tagging works in workload management, which uses index patterns as its primary attribute.
+
+### Multiple attribute matching
+
+```json
+{
+  "index_patterns": ["logs-prod-*"],
+  "request_type": ["search", "count"],
+  "workload_group": "production_search_workload_id"
+}
+
+{
+  "index_patterns": ["logs-prod-*"],
+  "workload_group": "production_workload_id"
+}
+```
+
+### Attribute specificity
+
+```json
+{
+  "index_patterns": ["logs-*"],
+  "workload_group": "general_workload_id"
+}
+
+{
+  "index_patterns": ["logs-prod-service-*"],
+  "workload_group": "prod_service_workload_id"
+}
+```
+
+## Best practices
+
+Follow these best practices for designing and operating rule-based auto-tagging:
+
+### Designing rules
+
+When creating rules, focus on building logical, specific configurations that support your workload and access patterns. Consider the following guidelines:
+
+* Identify the most relevant attributes for your use case.
+* Use specific attribute values for precise control.
+* Combine multiple attributes when appropriate.
+* Use consistent naming conventions.
+* Document attribute matching behavior.
+
+### Managing attributes
+
+Attribute selection and configuration significantly influence rule effectiveness. To manage attributes successfully, perform the following actions:
+
+* Understand each attribute’s matching behavior.
+* Start with the most specific criteria needed.
+* Avoid overlapping rules unless intentional.
+* Plan for future attribute value patterns.
+
+### Operations
+
+Ongoing operations and monitoring help maintain rule quality over time. Use the following practices to keep your setup reliable and effective:
+
+* Test new rules in a development environment.
+* Monitor rule matches in system logs.
+* Document rule configurations.
+* Regularly review rule effectiveness.
+* Remove unused rules.
+
+## Troubleshooting
```

Comment:
@kaushalmahi12: We need a little more detail for each of these items to justify keeping this section. For example, when I need to "Verify attribute values and match rules" because no attributes have been assigned, how would I verify those attributes? The rules API?

Reply:
Sure, I will expand on this.